lib/orderbook/OrderBook.ts

```diff
@@ -250,106 +144,77 @@ class OrderBook extends EventEmitter {
 
     const stampedOrder: orders.StampedPeerOrder = { ...order, createdAt: ms() };
 
-    if (!this.addOrder(this.peerOrders, stampedOrder)) {
+    if (!matchingEngine.addPeerOrder(stampedOrder)) {
       this.logger.debug(`incoming peer order is duplicated: ${order.id}`);
       // TODO: penalize peer
       return false;
     }
 
-    matchingEngine.addPeerOrder(stampedOrder);
-
     this.logger.debug(`order added: ${JSON.stringify(stampedOrder)}`);
     this.emit('peerOrder.incoming', stampedOrder);
 
     return true;
   }
 
-  private removePeerOrders = async (peer: Peer): Promise<void> => {
-    this.matchingEngines.forEach((matchingEngine) => {
-      const orders = matchingEngine.removePeerOrders(peer.nodePubKey!);
-
-      orders.forEach((order) => {
-        this.removeOrder(this.peerOrders, order.id, order.pairId);
-        this.emit('peerOrder.invalidation', {
-          orderId: order.id,
-          pairId: order.pairId,
-        });
-      });
-    });
-  }
-
-  private updateOrderQuantity = (order: orders.StampedOrder, quantityToDecrease: number) => {
-    const isOwnOrder = orders.isOwnOrder(order);
-    const orderMap = this.getOrderMap(isOwnOrder ? this.ownOrders : this.peerOrders, order);
+  public removeOwnOrderByLocalId = (pairId: string, localId: string): { removed: boolean, globalId?: string } => {
+    const orderId = this.localIdMap.get(localId);
 
-    const orderToUpdate = orderMap.get(order.id);
-    if (!orderToUpdate) {
-      return false;
-    }
-    orderToUpdate.quantity -= quantityToDecrease;
-    if (orderToUpdate.quantity === 0) {
-      if (isOwnOrder) {
-        const { localId } = order as orders.StampedOwnOrder;
-        this.localIdMap.delete(localId);
-      }
-      orderMap.delete(order.id);
+    if (orderId === undefined) {
+      return { removed: false, globalId: orderId };
```

Comment:
What about just:
```
    if (!orderId) {
      return { removed: false };
```

Reply:
Haven't touched this row either. I left it like this because I thought that it's a bit safer like that. But I can change it.